src/main/java/org/openstreetmap/atlas/geography/atlas/AbstractAtlas.java

```diff
@@ -164,6 +164,17 @@ public static Atlas forOsmPbf(final Resource resource)
         return Iterables.filterTranslate(areasIntersecting(polygon), item -> item, matcher);
     }
 
+    @Override
+    public Iterable<Area> areasWithin(final Polygon polygon)
+    {
+        final Iterable<Area> areas = this.getAreaSpatialIndex().get(polygon.bounds());
+        return Iterables.filter(areas, area ->
```

Comment:
Optional: This is the old way of filtering. How about bringing new changes with `Iterables.stream(...).filter(...)`?

Reply:
Saw this too, wanted to stay consistent to the rest of the class. Let me go ahead and re-write the rest of the function to follow the same format.